### PR TITLE
feat(edge): Adds ability to parse retention policy on delete series

### PR DESCRIPTION
**Pull request by devanbenz**
- Per: https://github.com/influxdata/feature-requests/issues/175 
- Removes influxQL parser from error'ing on a given retention policy 
- Previously a query like this would error with `retention policy not supported at line 1, char 1`
```
DELETE FROM "test17510year"."average_temperature" where time >= '2019-09-17T20:00:00Z' AND time <= '2019-09-17T21:42:00Z'
```
this was making it impossible to delete a series within a retention policy only.

This is part one of a change to influxdb that allows for series deletion outside of the default RP.

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -3548,7 +3548,6 @@ func TestParser_ParseStatement(t *testing.T) {
 		{s: `DELETE`, err: `found EOF, expected FROM, WHERE at line 1, char 8`},
 		{s: `DELETE FROM`, err: `found EOF, expected identifier at line 1, char 13`},
 		{s: `DELETE FROM myseries WHERE`, err: `found EOF, expected identifier, string, number, bool at line 1, char 28`},
-		{s: `DELETE FROM "foo".myseries`, err: `retention policy not supported at line 1, char 1`},
```
Comment (davidby-influx) [inline parser_test.go:3551]: Should this case be retained, but the expected value be changed to no error?

**Review by davidby-influx — CHANGES_REQUESTED** [review]
Why all the change in `.idea/`?  Is thgis going to affect everyone's IDEs?

One code comment in tests.

**Comment by devanbenz** [comment]
> Why all the change in `.idea/`? Is thgis going to affect everyone's IDEs?

This is expected. `.idea` has user specific workspace stuff in it AFAIK. Also taking a look at the `influxdb` core repo `.idea/` is gitignore'd.

File: parser_test.go (hunk 1 above)
Comment (devanbenz) [reply parser_test.go:3551]: Looking through this block of code it appears that its just testing for errors, I don't see any that don't test for errors here specifically.

Comment (davidby-influx) [reply parser_test.go:3551]: So let's add in some invalid retention policies.  Names with illegal characters, stuff like that.

**Review by davidby-influx — APPROVED** [review]
LGTM

**Pull request merged**